source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1180 @@
+          {
+            // get reference left and right end of this section
+            const auto Lj  = arclength_L[j];
+            const auto Ljp = arclength_L[j + 1];
+
+
+            while (Lj <= i * deltaX && i * deltaX <= Ljp &&
+                   i < number_points - 1)
+              {
+                double l = i * deltaX;
+
+                // perform linear interpolation
+                equidist[i] = Point<2>((l - Lj) / (Ljp - Lj) *
+                                         (non_equidistant_points[j + 1] -
+                                          non_equidistant_points[j]) +
+                                       non_equidistant_points[j]);
+                i++;
+              }
+          }
+        return equidist;
+      }
+
+
+
+      /**
+       * Create the coarse grid.
+       * Initializes a given triangulation with a coarse grid.
+       * Create 6 coarse grids based on points A-L (class fields) and merges
+       * them to one triangulation.
+       *
+       * @param tria
+       */
+      void make_coarse_grid(Triangulation<2> &tria) const
+      {
+        // create vector of serial triangulations for each block and temporal
+        // storage for merging them
+        std::vector<Triangulation<2>> trias(10);
+
+        // helper function to create a subdivided quadrilateral
+        auto make = [](Triangulation<2> &               tria,
+                       const std::vector<Point<2>> &    corner_vertices,
+                       const std::vector<unsigned int> &repetitions,
+                       const unsigned int               material_id) {
+          // create subdivided rectangle with corner points (-1,-1) and (+1, +1)
+          // it serves as reference system
+          GridGenerator::subdivided_hyper_rectangle(tria,
+                                                    repetitions,
+                                                    {-1, -1},
+                                                    {+1, +1});
+
+          // move all vertices to the correct position
+          for (auto it = tria.begin_vertex(); it < tria.end_vertex(); ++it)
+            {
+              auto &       point = it->vertex();
+              const double xi    = point(0);
+              const double eta   = point(1);
+
+              // bilinear mapping
+              point = 0.25 * ((1 - xi) * (1 - eta) * corner_vertices[0] +
+                              (1 + xi) * (1 - eta) * corner_vertices[1] +
+                              (1 - xi) * (1 + eta) * corner_vertices[2] +
+                              (1 + xi) * (1 + eta) * corner_vertices[3]);
+            }
+
+          // set material id of block
+          for (auto cell : tria.active_cell_iterators())
+            cell->set_material_id(material_id);
+        };
+
+        // create a subdivided quadrilateral for each block (see last number of
+        // block id)
+        make(trias[0], {A, B, D, C}, {n_subdivision_y, n_subdivision_x_0}, 1);
```

Comment:
Could you make all instances of block ids static constants. You could write at the top of the class:
```cpp
static const unsigned int id_block_1 = 1;
```
and write here:
```suggestion
        make(trias[0], {A, B, D, C}, {n_subdivision_y, n_subdivision_x_0}, id_block_1);
```
This would make it easier to distinguish block ids form actual numbers.

Reply:
at the top of the class means right after `void make_coarse_grid(Triangulation<2> &tria) const
      {` or as class lists of MeshGenerator class?